### PR TITLE
Making the event component

### LINKED ISSUE #112: Creating Event Component
### Description
<write your description> This component will contain
 - Event table component

Event Table component will be passed these data as props
* Different events sorted by date as Running/Upcoming/Past Events (refer screenshot attached below).
* Pagination should be there.

### Concerned Folder
https://github.com/houseofgeeks/hg/tree/front-end/client/src/components/Event

### Assets 
https://drive.google.com/drive/folders/10TEVnDrYLLe8nbkYQLwT2qysBq65PPq-
### Screenshots/testimonials
<place your screenshots or testimonials here if applicable>

event(2)

### LINKED ISSUE #115: Creating Event table
### Description
In ref to #112, Create a table component as described in the image.
Take dummy data and pass it through props.

### PR DESCRIPTION
**Pull request by EmperorYP7**
fixes #112 ~fixes#116~ fixes #115
<Do not delete this template>
Description: <provide a description> Made a tabular structure for displaying all the events.

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes which were demanded in the linked issue.
- [x] I've tested my code on a Chromium based browser.
- [x] I've tested my code on Mozilla Firefox.
- [x] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial

https://deploy-preview-191--houseofgeeks.netlify.app/event

image

### CODE REVIEW

**Review by tusharjain0022 — COMMENTED** [review]
LGTM ! Nice work @EmperorYP7 🤩 
Just a small change that in the date | time column as time is being stored acc. to 12-hr time convention then it is should be printed with AM or PM.

**Comment by ankiiitraj** [comment]
Add a column for description and use extrnal_link_icon after the title! 
rest LGTM
something like 

the package used for icons https://react-icons.github.io/react-icons/

Screenshot from 2020-10-29 01-44-20

File: client/src/components/Event/Tables/UpcomingEvents.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import "bootstrap/dist/css/bootstrap.min.css";
+import './UpcomingEvents.css';
+
```
Comment (anktrj) [inline client/src/components/Event/Tables/UpcomingEvents.js:5]: add 
` import { BiLinkExternal } from 'react-icons/bi'; `

File: client/src/components/Event/Tables/UpcomingEvents.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                            if (date >= currentDate) {
+                                return (
+                                    <tr key={id} >
+                                        <td className="event-title">
+                                            <a
+                                                rel="noopener noreferrer"
+                                                href={`${event.eventLink}`}
+                                                target="_blank"
+                                            >
+                                                {`${event.title}`}
```
Comment (anktrj) [inline client/src/components/Event/Tables/UpcomingEvents.js]: add 
`<BiLinkExternal/>`
here for the icon

**Review by ankiiitraj — COMMENTED** [review]
Also, try adding a message to show while the evnts are being fetched!

**Comment by ankiiitraj** [comment]
The columns are not aligned and look odd! What do you say @hariom1625 @tusharjain0022 
Screenshot from 2020-10-29 11-57-15

**Comment by hariom1625** [comment]
> The columns are not aligned and look odd! What do you say @hariom1625 @tusharjain0022

I too noticed it does look a bit odd and also the prior 24 hrs date and time format was alright because the present commit is quite confusing maybe it isn't rendering properly.
And if we will be using 24hrs format, printing AM and PM wouldn't be necessary.

**Comment by EmperorYP7** [comment]
> The columns are not aligned and look odd! What do you say @hariom1625 @tusharjain0022

I am working my way around it. If anyone figures it out, you're welcome to push in some code here. :wink:


> I too noticed it does look a bit odd and also the prior 24 hrs date and time format was alright because the present commit is quite confusing maybe it isn't rendering properly.
> And if we will be using 24hrs format, printing AM and PM wouldn't be necessary.

I've fixed it, will be pushing it soon. It was something wrong with the way I implemented the function to convert 24hr format to 12hr format which didn't work out as expected in firefox.

**Review by ankiiitraj — APPROVED** [review]

**Review by tusharjain0022 — APPROVED** [review]
LGTM! 🤩

**Comment by ankiiitraj** [comment]
@EmperorYP7 is it ready to merged?

**Comment by EmperorYP7** [comment]
image

I am noticing some wierd margin at the top. Couldn't figure out where its coming from. But this PR can be merged. We can make an issue to solve this.

File: client/src/components/Event/Event.css [hunk 3]
```diff
@@ -1,12 +1,47 @@
...
+    height: auto;
 }
 
-.event-component .event_title {
+.event-component .event-title {
     font-family: 'Raleway';
     color: #EE2A7B;
     font-size: 8vh;
     opacity: 1;
     margin: 5vh 0;
```
Comment (tusharjain0022) [inline client/src/components/Event/Event.css]: here the top margin of `.event-title` is causing that unnecessary margin on the top. it should be `margin-bottom: 5vh ;`

Comment (tusharjain0022) [reply client/src/components/Event/Event.css]: I would suggest that this issue can be added to #195 with other minor updates or we can open a separate issue for it with issue of the alignment of columns of the table. and this PR is ready to be merged as here's already a lot of work done.

**Review by hariom1625 — APPROVED** [review]
LGTM !!
Great work @EmperorYP7  :-)

File: client/src/components/Event/Event.css (hunk 3 above)
Comment (EmperorYP7) [reply client/src/components/Event/Event.css]: That seems to be working for now. Thanks for pointing out @tusharjain0022! :smile:

**Pull request merged**